src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -380,18 +380,30 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	{
 #if PG_VERSION_NUM >= PG_VERSION_16
 		DefElem *opt = NULL;
+		 bool appended_option = false; // To track if any option has been appended
```

Comment:
I think you don't need to track if any option has been appended, blank lines are inserted in future deparse lines either way. For example here there is a blank line in GRANTED BY:
```C
if (stmt->grantor)
{
   appendStringInfo(buf, " GRANTED BY %s",
      RoleSpecString(stmt->grantor, true));
}
```